webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/editablemapping/EditableMappingIOService.scala

```diff
@@ -0,0 +1,146 @@
+        edgesZarrHeader)
+      isAdditionHeaderStream = NamedFunctionStream.fromJsonSerializable(
+        s"$arrayNameEdgeIsAddition/${Zarr3ArrayHeader.FILENAME_ZARR_JSON}",
+        isAdditionZarrHeader)
+      groupHeader = EmptyZarr3GroupHeader()
+      groupHeaderStream = NamedFunctionStream.fromJsonSerializable(EmptyZarr3GroupHeader.FILENAME_ZARR_JSON,
+                                                                   groupHeader)
+      allStreams = edgesZarrChunksStream ++ isAdditionZarrChunksStream ++ Seq(edgesHeaderStream,
+                                                                              isAdditionHeaderStream,
+                                                                              groupHeaderStream)
+      tempFilePath = tempFileService.create(f"${tracingId}_editedMappingEdges")
+      outputStream = new BufferedOutputStream(new FileOutputStream(new File(tempFilePath.toString)))
+      _ <- ZipIO.zip(allStreams, outputStream, level = Deflater.BEST_SPEED)
+      _ = Instant.logSince(before, s"Exporting ${editedEdges.length} edited mapping edges to zipped zarr", logger)
+    } yield tempFilePath
+
+  private def edgeIsAdditionToZarrChunks(editedEdges: Seq[(Long, Long, Boolean)]): Iterator[Array[Byte]] =
+    editedEdges.grouped(ChunkSize).map { edgeTupleChunk: Seq[(Long, Long, Boolean)] =>
+      val bytes = ByteBuffer.allocate(ChunkSize)
+      edgeTupleChunk.foreach {
+        case (_, _, isAddedEdge) =>
+          val boolAsByte: Byte = if (isAddedEdge) 1 else 0
+          bytes.put(boolAsByte)
+      }
+      compressor.compress(bytes.array)
+    }
+
+  private def editedEdgesToZarrChunks(editedEdges: Seq[(Long, Long, Boolean)]): Iterator[Array[Byte]] =
+    editedEdges.grouped(ChunkSize).map { edgeTupleChunk: Seq[(Long, Long, Boolean)] =>
+      val bytes = ByteBuffer.allocate(2 * ChunkSize * 8)
+      edgeTupleChunk.foreach {
+        case (src, dst, _) =>
+          bytes.putLong(src)
+          bytes.putLong(dst)
+      }
+      compressor.compress(bytes.array)
+    }
+
+  private lazy val compressorConfiguration =
+    BloscCodecConfiguration(
+      BloscCompressor.defaultCname,
+      BloscCompressor.defaultCLevel,
+      StringCompressionSetting(BloscCodecConfiguration.shuffleSettingFromInt(BloscCompressor.defaultShuffle)),
+      Some(BloscCompressor.defaultTypesize),
+      BloscCompressor.defaultBlocksize
+    )
+
+  private lazy val compressor =
+    new BloscCompressor(
+      Map(
+        BloscCompressor.keyCname -> StringCompressionSetting(BloscCompressor.defaultCname),
+        BloscCompressor.keyClevel -> IntCompressionSetting(BloscCompressor.defaultCLevel),
+        BloscCompressor.keyShuffle -> IntCompressionSetting(BloscCompressor.defaultShuffle),
+        BloscCompressor.keyBlocksize -> IntCompressionSetting(BloscCompressor.defaultBlocksize),
+        BloscCompressor.keyTypesize -> IntCompressionSetting(BloscCompressor.defaultTypesize)
+      ))
```

Comment:
So the configs in `compressorConfiguration` and `compressor` always need to be kept in sync? Isn't is possible to just create the compressor and get its configuration? We coded this ourselves. So this might be something for a future refactoring :thinking:?